src/Symfony/Component/Serializer/Encoder/CsvEncoder.php

```diff
@@ -21,6 +21,10 @@
 class CsvEncoder implements EncoderInterface, DecoderInterface
 {
     const FORMAT = 'csv';
+    const DELIMITER_KEY = 'csv_delimiter';
```

Comment:
Perhaps your confusion is legit and the naming should be more explicit though. I named it after `DateTimeNormalizer::FORMAT_KEY`, but that's currently the only occurence of such a suffixed constant in the serializer component. Other constants for options like in the [AbstractNormalizer](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php#L30-L32) are simply named after the option itself, without any suffix.

Also in a different topic, some options like `allow_extra_attributes`, `attributes`, or `key_type` don't even have any constant. Maybe it's worth adding them and always ask for constants for new options in the future?

ping @dunglas.